src/Razor/test/Microsoft.VisualStudio.LanguageServerClient.Razor.Test/DocumentMapping/DefaultLSPDocumentMappingProviderTest.cs

```diff
@@ -0,0 +1,357 @@
+    private readonly Lazy<LSPDocumentManager> _documentManager;
+
+    public DefaultLSPDocumentMappingProviderTest(ITestOutputHelper testOutput)
+        : base(testOutput)
+    {
+        var csharpVirtualDocumentSnapshot = new CSharpVirtualDocumentSnapshot(s_razorVirtualCSharpFile, new StringTextSnapshot(string.Empty), hostDocumentSyncVersion: 0);
+        var documentSnapshot1 = new TestLSPDocumentSnapshot(s_razorFile, version: 1, "first doc", csharpVirtualDocumentSnapshot);
+        var documentSnapshot2 = new TestLSPDocumentSnapshot(s_anotherRazorFile, version: 5, "second doc", csharpVirtualDocumentSnapshot);
+        var documentManager = new TestDocumentManager();
+        documentManager.AddDocument(s_razorFile, documentSnapshot1);
+        documentManager.AddDocument(s_anotherRazorFile, documentSnapshot2);
+        _documentManager = new Lazy<LSPDocumentManager>(() => documentManager);
+        _razorLSPConventions = new RazorLSPConventions(TestLanguageServerFeatureOptions.Instance);
+    }
+
+    [Fact]
+    public async Task RazorMapToDocumentRangeAsync_InvokesLanguageServer()
+    {
+        // Arrange
+        var uri = new Uri("file:///some/folder/to/file.razor");
+
+        var response = new RazorMapToDocumentRangesResponse()
+        {
+            Ranges = new[] {
+                new Range()
+                {
+                    Start = new Position(1, 1),
+                    End = new Position(3, 3),
+                }
+            },
+            HostDocumentVersion = 1
+        };
+        var requestInvoker = new Mock<LSPRequestInvoker>(MockBehavior.Strict);
+        requestInvoker
+            .Setup(r => r.ReinvokeRequestOnServerAsync<RazorMapToDocumentRangesParams, RazorMapToDocumentRangesResponse>(
+                It.IsAny<ITextBuffer>(),
+                LanguageServerConstants.RazorMapToDocumentRangesEndpoint,
+                RazorLSPConstants.RazorLanguageServerName,
+                It.IsAny<Func<JToken, bool>>(),
+                It.IsAny<RazorMapToDocumentRangesParams>(),
+                It.IsAny<CancellationToken>()))
+            .ReturnsAsync(new ReinvocationResponse<RazorMapToDocumentRangesResponse>("TestLanguageClient", response));
+
+        var mappingProvider = new DefaultLSPDocumentMappingProvider(requestInvoker.Object, _documentManager, _razorLSPConventions);
+        var projectedRange = new Range()
+        {
+            Start = new Position(10, 10),
+            End = new Position(15, 15)
+        };
+
+        // Act
+        var result = await mappingProvider.MapToDocumentRangesAsync(RazorLanguageKind.CSharp, uri, new[] { projectedRange }, DisposalToken);
+
+        // Assert
+        Assert.NotNull(result);
+        Assert.Equal(1, result.HostDocumentVersion);
+        var actualRange = result.Ranges[0];
+        Assert.Equal(new Position(1, 1), actualRange.Start);
+        Assert.Equal(new Position(3, 3), actualRange.End);
+    }
+
+    //[Fact]
```

Comment:
Should these tests be skipped rather than commented out?

Reply:
Whoops, missed these! They should be deleted :D